packages/webapp/src/pages/editor.tsx

```diff
@@ -112,28 +115,31 @@ const Editor: FablePage = ({ decks, setDecks, isHydrated }) => {
 
   const handleSave = () => {
     if (!deckName.trim()) {
-      // Deck name is empty or only whitespace
       setIsDeckNameValid(false);
-      return; // Prevent further actions
+      return;
     }
     setIsDeckNameValid(true);
   
-    // Check if a deck with the same name already exists
-    const existingDeckIndex = decks.findIndex(d => d.name === deckName);
-
+    let updatedDecks = [...decks];
+  
+    // Check if editing an existing deck and the name has changed
+    if (originalDeckIndex !== null && decks[originalDeckIndex].name !== deckName) {
+      // Remove the old deck
+      updatedDecks.splice(originalDeckIndex, 1);
+    }
+  
+    // Find if a deck with the new name already exists
+    const existingDeckIndex = updatedDecks.findIndex(d => d.name === deckName);
+  
     if (existingDeckIndex !== -1) {
-      // Deck with the same name exists, replace it
-      const updatedDecks = [...decks];
+      // Replace existing deck with the new name
       updatedDecks[existingDeckIndex] = { name: deckName, cards: deck };
-      setDecks(updatedDecks);
     } else {
-      // Create a new deck object
-      const newDeck = { name: deckName, cards: deck };
-    
-      // Add the new deck to the decks state
-      setDecks(prevDecks => [...prevDecks, newDeck]);
+      // Add as a new deck
+      updatedDecks.push({ name: deckName, cards: deck });

```

Comment:
We could avoid mutation here and use sthing like
```
let updatedDecks: Decks[] = []
...
// first branch
decks.toSpliced(originalsDeckIndex,1)
...
// second branch
decks.toSpliced(originalDeckIndex, 1, { name: deckName, cards: deck })
```

The big advantage would be avoiding changing the deck ordering.
Anyway, let's not do that, as we'll need to manage the decks on chain anyway :D